draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1083,63 +1083,43 @@ This section specifies such common definitions.
 
 ## Raw packet and frame information {#raw-info}
 
-While qlog is a more high-level logging format, it also allows the inclusion of
-most raw wire image information, such as byte lengths and even raw byte values.
-This can be useful when for example investigating or tuning packetization
-behavior or determining encoding/framing overheads. However, these fields are not
-always necessary and can take up considerable space if logged for each packet or
-frame. They can also have a considerable privacy and security impact. As such,
-they are grouped in a separate optional field called "raw" of type RawInfo (where
-applicable).
+While qlog is a high-level logging format, it also allows the inclusion of most
+raw wire image information, such as byte lengths and byte values. This is useful
+when for example investigating or tuning packetization behavior or determining
+encoding/framing overheads. However, these fields are not always necessary, can
+take up considerable space and can have a considerable privacy and security
+impact. Where applicable, these fields are grouped in a separate optional field
+named "raw" of type RawInfo. The exact definition of entities, headers, trailers
+and payloads depend on the protocol used.
 
 Definition:
 
 ~~~ cddl
 RawInfo = {
     ; the full byte length of the entity (e.g., packet or frame),
-    ; including headers and trailers
+    ; including possible headers and trailers
     ? length: uint64
 
     ; the byte length of the entity's payload,
-    ; without headers or trailers
+    ; excluding possible headers or trailers
     ? payload_length: uint64
 
-    ; the contents of the full entity,
-    ; including headers and trailers
+    ; the (potentially truncated) contents of the full entity,
+    ; including headers and possibly trailers
     ? data: hexstring
 }
 ~~~
 {: #raw-info-def title="RawInfo definition"}
 
-Note:
-
-: The RawInfo:data field can be truncated for privacy or security
-purposes (for example excluding payload data), see {{truncated-values}}.
-In this case, the length properties should still indicate the
-non-truncated lengths.
-
-Note:
-
-: We do not specify explicit header_length or trailer_length fields. In
-most protocols, header_length can be calculated by subtracting the payload_length
-from the length (e.g., if trailer_length is always 0). In protocols with trailers
-(e.g., QUIC's AEAD tag), event definitions documents SHOULD define other ways of
-logging the trailer_length to make the header_length calculation possible.
-
-: The exact definitions entities, headers, trailers and payloads depend on the
-protocol used. If this is non-trivial, event definitions documents SHOULD include
-a clear explanation of how entities are mapped into the RawInfo structure.
-
-Note:
+The RawInfo:data field can be truncated for privacy or security purposes, see
+{{truncated-values}}. In this case, the length and payload_lenght fields should
```

Comment:
is this a SHOULD requirement?

Either way, it might help to explain why keeping the non-truncated length is advised.